Updating developer documentation

PrasadBabarendaGamage opened the pull request:
Adding updates required to address issue #65 (includes updates from Richard's review of pull request #66)

PrasadBabarendaGamage commented:
Not all of Richards reviewer comments from pull request #66 have been resolved - work in progress.

hsorby commented:
What's with the manual line wrapping?

PrasadBabarendaGamage commented on contribute/dev_contribution.rst:
```diff
@@ -76,25 +82,30 @@ It may be necessary to refactor the current design to enable the easiest possibl
+===================
 
-The ``git add`` and ``git commit`` commands should be obvious,  the ``git push`` command sets the local branch ``issue46`` to be linked with the remote branch ``issue46`` in the origin (the default shorthand for your libCellML repository on GitHub) repository, this branch will be created in the origin repository if it doesn't already exist.
+From there create a pull request from your feature branch to the :term:`prime repository` develop branch. When creating the pull request make sure to add in the comment ``addresses issue #46`` (replace the number 46 with the actual number of the issue you are addressing), or something to that effect. This will create a link between the issue and the pull request enabling other people to see that  ...
```
@nickerso 
Should addresses issue #46 also be used when making commits (not just a pull request), to make sure the commit you are making is linked to an issue? This is not mentioned in the "Committing changes to your feature branch" section.

nickerso replied:
commits on the same branch will automatically update the pull request, so I don't see a need for mentioning it in every commit. If referencing a different issue, then it makes sense to reference it - but its enough to have #46 in whatever context it needs to be. Its also trivial to reference issues in other repos when required.

PrasadBabarendaGamage replied:
Hi Andre, committing to the same branch without mentioning the issue in the commit message would update the pull. However, the issue itself would not get updated - this seems important as people won't know what work has been done if they only browse the issues.

nickerso replied:
This isn't a show stopper for me, either way.

PrasadBabarendaGamage commented:
Hi all, this pull request is ready. It addresses the issues Richard mentioned in pull request #66 (I've made some comments there regarding the updates). Items that have not been addressed have been added to new issues. If you have any suggestions for major improvements then please update the appropriate issues/add new issues. Please also indicate priority in the issue. Due to time constraints, I unfortunately will only be able to work on high priority issues. Will aim to work on adding some tutorials next based on e.g. Andre's monodomain example, and add documentation to help others do (see issue #67 for more information).

PrasadBabarendaGamage commented:
@hsorby @nickerso @chrispbradley Hi all, hoping if we could please merge this pull request in as soon as possible. I can then point the developers to the staging site to have a look. We created a list of who is working on what in OpenCMISS and there are columns for issues and pull requests. Would be great if developers can see what is the recommend workflow for adding these items. This pull request provides the latest info for explaining how to do this.

PrasadBabarendaGamage commented:
@hsorby @nickerso @chrispbradley Hi guys, Andy just mentioned that the Stuttgart developers are having a developer meeting tomorrow and he was going to mention the developer docs - it would be great if this could be pulled in and up on staging by then so that they can have a look and comment/provide feedback.
I think in future it might be best if we can just push to staging prior to review so that reviewers can compare with the non-staging site. We then also have a working link to the latest docs on staging.

Thanks

PrasadBabarendaGamage commented:
@nickerso @hsorby  Thanks a lot guys!
